src/setupWorker/start/createStart.ts

```diff
@@ -59,14 +63,21 @@ export const createStart = (context: ComposeMocksInternalContext) => {
       context.worker = worker
       context.registration = registration
 
-      window.addEventListener('beforeunload', () => {
+      const beforeUnload = () => {
         if (worker.state !== 'redundant') {
           // Notify the Service Worker that this client has closed.
           // Internally, it's similar to disabling the mocking, only
           // client close event has a handler that self-terminates
           // the Service Worker when there are no open clients.
           worker.postMessage('CLIENT_CLOSED')
         }
+      }
+      window.addEventListener('beforeunload', beforeUnload)
```

Comment:
If we adopt the `context.addListener()` pattern, this can be simplified, as we won't have to repeat the `beforeUnload` twice.

Reply:
I have only this doubt, why you said that we do this twice?